forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2022 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth
+
+import android.content.Context
+import android.content.SharedPreferences
+
+interface ConfigInterface {
+    fun loadFromPreference(context: Context?): FROptions?
+}
+
+class ConfigHelper: ConfigInterface {
+
+    companion object {
+
+        const val ORG_FORGEROCK_V_1_HOSTS = "org.forgerock.v1.HOSTS"
+
+        @JvmStatic
+        fun persist(context: Context, frOptions: FROptions) {
+            val sharedPreferences =
+                context.getSharedPreferences(ORG_FORGEROCK_V_1_HOSTS, Context.MODE_PRIVATE)
+            sharedPreferences.edit().putString("url", frOptions.server.url)
+                .putString("realm", frOptions.server.realm)
+                .putString("cookieName", frOptions.server.cookieName)
+                .putString("client_id", frOptions.oauth.oauthClientId)
+                .putString("revoke_endpoint", frOptions.urlPath.revokeEndpoint)
+                .putString("end_session_endpoint", frOptions.urlPath.endSessionEndpoint).apply()
+        }
+
+        @JvmStatic
+        fun isConfigChanged(context: Context, frOption: FROptions): Boolean {
+            val sharedPreferences =
+                context.getSharedPreferences(ORG_FORGEROCK_V_1_HOSTS, Context.MODE_PRIVATE)
+            sharedPreferences.getString("url", null)?.apply {
+                if(frOption.server.url != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("realm", null)?.apply {
+                if(frOption.server.realm != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("client_id", null)?.apply {
+                if(frOption.oauth.oauthClientId != this) {
+                    return true
+                }
+            }
+            sharedPreferences.getString("cookieName", null)?.apply {
+                if(frOption.server.cookieName != this) {
+                    return true
+                }
+            }
+            return false
+        }
+
+        @JvmStatic
+        fun load(context: Context, frOption: FROptions?): FROptions {
+            return frOption?.let {
+                FROptionsBuilder.build {
```

Comment:
Any reason we cannot just return the input frOption, but need to clone it?

Reply:
@witrisna  this is a perfect catch,  i was moving few things, yes we need to return as it is.